src/Kafka/ConsumerGroups/components/ConsumerGroupResetOffset.tsx

```diff
@@ -0,0 +1,274 @@
+import {
+  Alert,
+  Button,
+  Checkbox,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  Stack,
+  StackItem,
+  TextInput,
+} from "@patternfly/react-core";
+import type { FunctionComponent } from "react";
+import { useTranslation } from "react-i18next";
+import "../ConsumerGroup.css";
+import type { Consumer, OffsetValue } from "../types";
+import {
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@patternfly/react-table";
+import { TopicSelect } from "./TopicSelect";
+import { OffsetSelect } from "./OffsetSelect";
+
+export type ConsumerRow = Consumer & {
+  selected?: boolean;
+};
+
+export type ConsumerGroupResetOffsetProps = {
+  isModalOpen: boolean;
+  isDisconnected: boolean;
+  groupId: string;
+  topics: string[];
+  selectedTopic: string;
+  customOffsetValue: string;
+  setcustomOffsetValue: (customOffsetValue: string) => void;
+  consumers: ConsumerRow[];
+  isSelected: boolean;
+  onClickClose: () => void;
+  onClickResetOffset: () => void;
+  onChangeTopic: (value: string) => void;
+  selectedOffset: OffsetValue;
+  onChangeOffsetValue: (value: OffsetValue) => void;
+  confirmCheckboxChecked: boolean;
+  onConfirmationChange: (value: boolean) => void;
+  SelectAllConsumer: (consumer: ConsumerRow[]) => void;
+  onSelectAll: (isSelected: boolean) => void;
+  onSelectRow: (index: number, isSelected: boolean) => void;
+  disableFocusTrap?: boolean;
+  appendTo?: () => HTMLElement;
+};
+
+export const ConsumerGroupResetOffset: FunctionComponent<
+  ConsumerGroupResetOffsetProps
+> = ({
+  isDisconnected,
+  groupId,
+  topics,
+  customOffsetValue,
+  setcustomOffsetValue,
+  consumers,
+  onClickClose,
+  isModalOpen,
+  onClickResetOffset,
+  selectedTopic,
+  onChangeTopic,
+  selectedOffset,
+  onChangeOffsetValue,
+  confirmCheckboxChecked,
+  onConfirmationChange,
+  isSelected,
+  onSelectAll,
+  onSelectRow,
+  disableFocusTrap,
+  appendTo,
+}) => {
+  const { t } = useTranslation(["kafka"]);
+
+  const isResetOffsetDisabled = (): boolean => {
+    return (
+      !selectedTopic ||
+      !selectedOffset ||
+      !confirmCheckboxChecked ||
+      !isDisconnected ||
+      consumers.filter(({ selected }) => selected === true).length === 0
+    );
+  };
+
+  const tableColumns = {
+    partition: t("consumerGroup.partition"),
+    clientId: `${t("consumerGroup.client_id")} + ${t(
+      "consumerGroup.member_id"
+    )}`,
+    current_offset: t("consumerGroup.current_offset"),
+    log_end_offset: t("consumerGroup.log_end_offset"),
+    offset_lag: t("consumerGroup.offset_lag"),
+    new_offset: t("consumerGroup.new_offset"),
+  };
+
+  return (
+    <Modal
```

Comment:
I'd suggest top-aligning the modal. We noticed in the create Kafka one that if the modal's content is dynamic if it's not top-aligned, it moves on the screen. It's disorientating.